include/ignition/gui/qml/GzPose.qml

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+import QtQuick 2.9
+import QtQuick.Controls 2.2
+import QtQuick.Controls.Material 2.1
+import QtQuick.Layouts 1.3
+import QtQuick.Controls.Styles 1.4
+
+/**
+ *  Item displaying 3D pose information.
+ *
+ *  Users should load values to xValues, yValues, etc.
+ *  If readOnly == False,
+ *  users can read from signal pararmeters of GzPoseSet: _x, _y, etc.
+ *
+ *  Usage example:
+ *  GzPose {
+ *    id: gzPose
+ *    readOnly: false
+ *    xValue: xValueFromCPP
+ *    yValue: yValueFromCPP
+ *    zValue: zValueFromCPP
+ *    rollValue: rollValueFromCPP
+ *    pitchValue: pitchValueFromCPP
+ *    yawValue: yawValueFromCPP
+ *    onGzPoseSet: {
+ *      myFunc(_x, _y, _z, _roll, _pitch, _yaw)
+ *    }
+ *  }
+**/
+
+Item {
+  id: gzPoseRoot
+
+  // Read-only / write
+  property bool readOnly: false
+
+  // User input value
+  property double xValue
+  property double yValue
+  property double zValue
+  property double rollValue
+  property double pitchValue
+  property double yawValue
+
+  /**
+   * Useful only when readOnly == false. User should read spinbox values from
+   * its parameters.
+   */
+  signal gzPoseSet(double _x, double _y, double _z, double _roll, double _pitch, double _yaw)
+
+
+  /*** The following are private variables: ***/
+  // Show Pose bar (used to control expand)
+  property bool show: true
+
+  height: gzPoseContent.height
+
+  // Left indentation
+  property int indentation: 10
+
+  // Horizontal margins
+  property int margin: 5
+
+  // Maximum spinbox value
+  property double spinMax: 1000000
+
+  // local variables to store spinbox values
+  property var xItem: {}
+  property var yItem: {}
+  property var zItem: {}
+  property var rollItem: {}
+  property var pitchItem: {}
+  property var yawItem: {}
+
+  // Dummy component to use its functions.
+  IgnHelpers {
+    id: gzHelper
+  }
```

Comment:
Well. This is doable but we need to make sure every qml using this common widget must also include `IgnHelpers.qml` as well. Do we want to do this?

Reply:
Agreed that this would be a nice change. This PR is just following the weird pattern that I started with `IgnHelpers` a while back, so we don't need to block on it.

> we need to make sure every qml using this common widget must also include IgnHelpers.qml as well.

I think there may be ways to work around this, either with a [singleton](https://doc.qt.io/qt-5/qtqml-modules-qmldir.html) or a [javascript file](https://doc.qt.io/qt-6/qtqml-javascript-expressions.html#functions-defined-in-a-javascript-file).